tests/test_sources.py

```diff
@@ -84,3 +85,9 @@ def test_default_download_source_with_predefined_url(
     download_source_check.assert_called_with(
         tmp_context.sdists_downloads, "predefined_url-1.0", "foo-1.0"
     )
+
+
+def test_invalid_version():
+    version = "fake version 1.0"
+    with pytest.raises(InvalidVersion):
+        Version(version)
```

Comment:
This is testing the `Version` object from the `packaging` library which we have no control over anyways. It will make more sense to test if `download_source` throws an error if it receives an invalid version from the plugin or the default function

Reply:
okay, will do